plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/DialogStagingDirectoryProvider.java

```diff
@@ -0,0 +1,27 @@
+package com.google.cloud.tools.eclipse.appengine.deploy.standard;
+
+import org.eclipse.swt.widgets.DirectoryDialog;
+import org.eclipse.swt.widgets.Shell;
+
+import com.google.cloud.tools.eclipse.appengine.deploy.Messages;
+
+/**
+ * Prompts the user to select a directory for staging.
+ */
+// TODO should be moved to separate UI bundle (if kept at all in deploy's final implementation)
```

Comment:
If this is to be moved to a separate bundle, then the interface should just be `DirectoryProvider`, and this should be a `DialogDirectoryProvider` with the labels provided as arguments.  After all, how many places are we going to need a staging directory?